docs/_source/docs/template_handlers.rst

```diff
@@ -0,0 +1,185 @@
+Template Handlers
+=================
+
+Template handlers can be used to resolve a ``template`` config block to a CloudFormation template. This can be used to
+fetch templates from S3, for example. Users can create their own template handlers to easily add support for other
+template loading mechanisms. See `Custom Template Handlers`_ for more information.
+
+When a ``template_path`` property is specified in the Stack config, it is wired into a ``file`` template handler by
+default. This saves you from having to specify a full ``template`` block if you just want to load a file from disk.
+Sceptre implements resolvers, which can be used to resolve a value of a
+CloudFormation ``parameter`` or ``sceptre_user_data`` value at runtime. This is
+most commonly used to chain the outputs of one Stack to the inputs of another.
+
+Syntax:
+
+.. code-block:: yaml
+
+   template:
+     type: s3
+     path: <bucket>/<key>
+
+Available Template Handlers
+---------------------------
+
+file
+~~~~~~~~~~~~~~~~~~~~
+
+Loads a template from disk. Supports JSON, YAML, Jinja2 and Python files. Will be used if the ``template_path`` Stack
+config property is set, for backwards compatibility reasons.
+
+Syntax:
+
+.. code-block:: yaml
+
+   template:
+     type: file
+     path: <path>
+
+Example:
+
+.. code-block:: yaml
+
+   template:
+     type: file
+     path: storage/bucket.yaml
+
+s3
+~~~~~~~~~~~~~
+
+Downloads a template from an S3 bucket. The template will be kept in memory and not be persisted to disk. Currently
+only supports YAML / JSON CloudFormation templates,
```

Comment:
hmm, we have a bunch of shared jinja2 templates in S3 buckets.  This won't help me with those now but I guess it's a good start.

Reply:
This is something that I want to fix in a next PR. We can find a way to share / extract the code that loads Jinja / Python templates and make it available to all template handlers. We do need to add a flag or type hint to the template handlers properties in that case, as some template handlers might not have the concept of a file extension or something to tell what type of template it is.